game/final_obs_gen.py

```diff
@@ -0,0 +1,494 @@
+                    [0, 0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 1, 0, 0],
+                ]
+            )
+        elif self.obs_type == "b":
+            self.ARRAY = np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0],
+                ]
+            )
+        else:
+            self.ARRAY = np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0],
+                ]
+            )
+
+    def setup_parameters(self):
+        self.height, self.width = self.ARRAY.shape
+        if self.obs_type in ("c", "C"):
+            self.y = self.REF_AXIS - self.height * self.GRID_SIZE
+        elif self.obs_type == "b":
+            self.y = self.LOW_BIRD_AXIS - self.height * self.GRID_SIZE
+        else:
+            self.y = self.HIGH_BIRD_AXIS - self.height * self.GRID_SIZE
+
+
+class ObstacleInteraction(Game):
+
+    def __init__(self):
+        super().__init__()
+
+        self.curr_cooldown = 0  # generates after however long (IN PIXELS)
+        self.cooldown_variance = np.linspace(10, 50, 5)  # variance to cooldown time
+        self.base_cooldown = 40  # must wait this long before new cacti generates
+        self.curr_obs = []  # obstacles generated for the game
+        self.possible_speeds = [10]
+        self.possible_objects = ["c", "C", "ccc", "CC", "cCc", "cc", "b", "B"]

```

Comment:
Some of these are constants. We can use all caps to indicate that. (It is quite unfortunate that Python does not have consts, so we have to resort to expressing it with naming conventions)

Reply:
Changed the constants here to match correctly